### DISCUSSION TITLE
Tools and apps for a large music library

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#7069] [Show and tell]
I’ve often been asked which tools and software I use to handle a large, well-tagged music library. Well, here’s the answer, with some links: https://blog.syvi.net/tech-snippets/tools-and-apps-for-a-large-music-library

**Comment by blindndangerous** [comment]
Thanks for this! I grabbed loudgain and am having it run through my files with rgbpm as I did use mp3gain a long long time ago on my collection. I use beets instead of picard but from what I've read they seem to be personal preference. I did notice that I do need to remove the -L switch from rgbpm on mp3 and m4a files, as if I don't beets will try to write 0 in it's database for RG_album values. Still with -L and -I3 removed it still wants to put rg_track_gain as 0.0.

**Reply by Moonbase59** [reply]
It’s a mess. Still. Modern software should treat RG tags case-insenitively, but often doesn’t. RG1/2 specs define UPPERCASE for ID3v2 tags, but since some programs _require_ lowercase I introduced the `-L` option. For instance, VLC only respects uppercase, IDJC and KODI only respect lowercase. Don’t know which kind beets needs, the docs are unclear. Maybe some experimentation with a few files needed… Also check if you’re using the RG plugin with beets, which might introduce a complete new set of problems.

Be absolutely sure you don’t have _both_ uppercase and lowercase variants in the tags! This is technically possible (at least with MP3), but will confuse almost all readers. _loudgain_ usually takes care of that, if you don’t use another RG tool after it.

More on the upper-/lowercase issue here: https://github.com/Moonbase59/loudgain?tab=readme-ov-file#uppercase-or-lowercase-replaygain_-tags

Personally, I’d use, in the beets configuration file:

```
import:
  write: no

replaygain:
  auto: no
```

which should keep beets from writing anything to my files, and never do a replaygain calculation itself (because loudgain does it much better).

**Reply by Marcioghiraldelli** [reply]
Hello, is it possible to use it to normalize Azuracast folders for more than one station just by indicating the path and the system can normalize the audio from folders and subfolders?

example:
script.sh /path/station/media/folders/subfolders

I searched and didn't find a way to do this here.

**Reply by Moonbase59** [reply]
Well, you could possibly use `rgbpm` for that, as in

```
rgbpm path/to/folder1 path/to/folder2 ...
```

See https://github.com/Moonbase59/loudgain/tree/master?tab=readme-ov-file#mass-tagging

You’d need a bash shell and loudgain installed for that, but I recommend doing all that pre-processing _before_ uploading music to a station.

**Comment by blindndangerous** [comment]
Yeah, I'll have to look around. For now, I've removed -L and -I3 and it's happily scanning along. I'll just take the missing values for now in the beets database, maybe it'll be fixed later.  I did use replaygain in my beets setup actually, this was before I knew Loudgain was a thing. It was using ffmpeg as it's backend though.
